From 6a5f70179d2b5a938b2253b4d77d690af2d16712 Mon Sep 17 00:00:00 2001
From: UtsavAgarwalADI <utsav.agarwal@analog.com>
Date: Wed, 19 Jun 2024 10:54:20 +0100
Subject: [PATCH 1/3] Adding support for SC598-SOM Rev E

- Updating qspi partitions to account for bigger flash storage
- Adding spi flash entry for chip compatibility
- Updating device tree to reflect new i2c gpio expander on the SOM

Signed-off-by: Utsav Agarwal <utsav.agarwal@analog.com>
---
 arch/arm64/boot/dts/adi/sc598-som.dtsi       | 75 ++++++++++----------
 arch/arm64/configs/sc598-som-ezkit_defconfig |  1 +
 drivers/mtd/spi-nor/issi.c                   |  4 ++
 3 files changed, 44 insertions(+), 36 deletions(-)

diff --git a/arch/arm64/boot/dts/adi/sc598-som.dtsi b/arch/arm64/boot/dts/adi/sc598-som.dtsi
index 92749e6cf9eb..9d1d4fc4a194 100644
--- a/arch/arm64/boot/dts/adi/sc598-som.dtsi
+++ b/arch/arm64/boot/dts/adi/sc598-som.dtsi
@@ -204,10 +204,10 @@ &spi2 {
 	status = "okay";
 	cs-gpios = <&gpa 5 GPIO_ACTIVE_LOW>;
 
-	flash: is25lp512@0 {
+	flash: is25lp01g@0 {
 		#address-cells = <1>;
 		#size-cells = <1>;
-		compatible = "is25lp512", "jedec,spi-nor";
+		compatible = "is25lp01g", "jedec,spi-nor";
 		reg = <0>;
 		spi-cpha;
 		spi-cpol;
@@ -236,7 +236,7 @@ qspi_3: partition@3 {
 
 		qspi_4: partition@4 {
 			label = "JFFS2 Formatted RFS";
-			reg = <0x1020000 0x2FE0000>;
+			reg = <0x1020000 0x6FE0000>;
 		};
 
 	};
@@ -255,11 +255,11 @@ &i2c2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c2_pins>;
 
-	ssw0: gpio@0x20 {
-		compatible = "microchip,mcp23018";
+	ssw0: gpio@0x34 {
+		compatible = "adi,adp5588-gpio";
 		gpio-controller;
 		#gpio-cells = <2>;
-		reg = <0x20>;
+		reg = <0x34>;
 		status = "okay";
 
 		pinctrl-names = "default";
@@ -271,70 +271,73 @@ ssw0pullups: pinmux {
 				"gpio4", "gpio5", "gpio6", "gpio8", "gpio9";
 		};
 
-		led1 {
-			gpio-hog;
-			gpios = <0 0x0>;
-			output-high;
-			line-name = "led1-en";
-		};
-
-		led2 {
+		uart0 {
 			gpio-hog;
-			gpios = <1 0x0>;
-			output-high;
-			line-name = "led2-en";
+			gpios = <0 GPIO_ACTIVE_HIGH>;
+			output-low;
+			line-name = "~uart0-en";
 		};
 
-		led3 {
+		uart0-flow-en {
 			gpio-hog;
-			gpios = <2 0x0>;
+			gpios = <1 GPIO_ACTIVE_HIGH>;
 			output-high;
-			line-name = "led3-en";
+			line-name = "~uart0-flow-en";
 		};
 
 		spi2d2-d3 {
 			gpio-hog;
-			gpios = <3 0x0>;
+			gpios = <2 GPIO_ACTIVE_HIGH>;
 			output-low;
 			line-name = "~spi2d2-d3-en";
 		};
 
 		spi2flash-cs {
 			gpio-hog;
-			gpios = <4 0x0>;
+			gpios = <3 GPIO_ACTIVE_HIGH>;
 			output-low;
 			line-name = "~spi2flash-cs";
 		};
 
-		uart0 {
+		emmc {
 			gpio-hog;
-			gpios = <5 0x0>;
+			gpios = <8 GPIO_ACTIVE_HIGH>;
 			output-low;
-			line-name = "~uart0-en";
+			line-name = "emmc-en";
 		};
 
-		uart0-flow-en {
+		emmc-som-en {
 			gpio-hog;
-			gpios = <6 0x0>;
+			gpios = <9 GPIO_ACTIVE_HIGH>;
 			output-high;
-			line-name = "~uart0-flow-en";
+			line-name = "emmc-som-en";
+		};
+		
+		led1 {
+			gpio-hog;
+			gpios = <15 GPIO_ACTIVE_HIGH>;
+			output-low;
+			line-name = "led1-en";
 		};
 
-		emmc {
+		led2 {
 			gpio-hog;
-			gpios = <8 0x0>;
+			gpios = <16 GPIO_ACTIVE_HIGH>;
 			output-low;
-			line-name = "emmc-en";
+			line-name = "led2-en";
 		};
 
-		emmc-som-en {
+		led3 {
 			gpio-hog;
-			gpios = <9 0x0>;
-			output-high;
-			line-name = "emmc-som-en";
+			gpios = <17 GPIO_ACTIVE_HIGH>;
+			output-low;
+			line-name = "led3-en";
 		};
+		
+	};
+
+
 	};
-};
 
 &mmc0{
 	pinctrl-names = "default";
diff --git a/arch/arm64/configs/sc598-som-ezkit_defconfig b/arch/arm64/configs/sc598-som-ezkit_defconfig
index b7ea7bab5dcb..34bc87aae3a9 100644
--- a/arch/arm64/configs/sc598-som-ezkit_defconfig
+++ b/arch/arm64/configs/sc598-som-ezkit_defconfig
@@ -218,6 +218,7 @@ CONFIG_PINCTRL_MCP23S08=y
 CONFIG_PINCTRL_SINGLE=y
 CONFIG_PINCTRL_ADSP_SC5XX=y
 CONFIG_SRUCTRL_ADSP_SC5XX=y
+CONFIG_GPIO_ADP5588=y
 CONFIG_GPIO_ADI_ADSP_PORT=y
 CONFIG_POWER_RESET_SYSCON=y
 CONFIG_SYSCON_REBOOT_MODE=y
diff --git a/drivers/mtd/spi-nor/issi.c b/drivers/mtd/spi-nor/issi.c
index ed7ead8bb7b1..6469876d21be 100644
--- a/drivers/mtd/spi-nor/issi.c
+++ b/drivers/mtd/spi-nor/issi.c
@@ -52,6 +52,10 @@ static const struct flash_info issi_parts[] = {
 			     SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ |
 			     SPI_NOR_4B_OPCODES)
 		.fixups = &is25lp256_fixups },
+	{ "is25lp01g",  INFO(0x9d601b, 0, 128 * 1024, 1024,
+			     SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ |
+			     SPI_NOR_4B_OPCODES)
+		.fixups = &is25lp256_fixups },
 	{ "is25wp032",  INFO(0x9d7016, 0, 64 * 1024,  64,
 			     SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
 	{ "is25wp064",  INFO(0x9d7017, 0, 64 * 1024, 128,
-- 
2.25.1

